TypeError on Python 3.12b4

```
tests/integration/test_schema_path.py:7: in <module>
    import responses
        BytesIO    = <class '_io.BytesIO'>
        Path       = <class 'pathlib.Path'>
        __builtins__ = <builtins>
        __cached__ = '/home/runner/work/jsonschema-spec/jsonschema-spec/tests/integration/__pycache__/test_schema_path.cpython-312.pyc'
        __doc__    = None
        __file__   = '/home/runner/work/jsonschema-spec/jsonschema-spec/tests/integration/test_schema_path.py'
        __loader__ = <_pytest.assertion.rewrite.AssertionRewritingHook object at 0x7fb1cfa9ae70>
        __name__   = 'test_schema_path'
        __package__ = ''
        __spec__   = ModuleSpec(name='test_schema_path', loader=<_pytest.assertion.rewrite.AssertionRewritingHook object at 0x7fb1cfa9ae70>, origin='/home/runner/work/jsonschema-spec/jsonschema-spec/tests/integration/test_schema_path.py')
        dumps      = <function dumps at 0x7fb1cfb9a7a0>
        mock       = <module 'unittest.mock' from '/opt/hostedtoolcache/Python/3.12.0-beta.3/x64/lib/python3.12/unittest/mock.py'>
        pytest     = <module 'pytest' from '/home/runner/work/jsonschema-spec/jsonschema-spec/.venv/lib/python3.12/site-packages/pytest/__init__.py'>
.venv/lib/python3.12/site-packages/responses/__init__.py:40: in <module>
    from typing_extensions import Literal
        Any        = typing.Any
        Callable   = typing.Callable
        ConnectionError = <class 'requests.exceptions.ConnectionError'>
        Dict       = typing.Dict
        FirstMatchRegistry = <class 'responses.registries.FirstMatchRegistry'>
        HTTPAdapter = <class 'requests.adapters.HTTPAdapter'>
        Iterable   = typing.Iterable
        Iterator   = typing.Iterator
        List       = typing.List
        Mapping    = typing.Mapping
        MaxRetryError = <class 'urllib3.exceptions.MaxRetryError'>
        Optional   = typing.Optional
        Pattern    = <class 're.Pattern'>
        RetryError = <class 'requests.exceptions.RetryError'>
        Sequence   = typing.Sequence
        Sized      = typing.Sized
        TYPE_CHECKING = False
        Tuple      = typing.Tuple
        Type       = typing.Type
        Union      = typing.Union
        _ThreadingLock = <built-in function allocate_lock>
        __builtins__ = <builtins>
        __cached__ = '/home/runner/work/jsonschema-spec/jsonschema-spec/.venv/lib/python3.12/site-packages/responses/__pycache__/__init__.cpython-312.pyc'
        __doc__    = None
        __file__   = '/home/runner/work/jsonschema-spec/jsonschema-spec/.venv/lib/python3.12/site-packages/responses/__init__.py'
        __loader__ = <_frozen_importlib_external.SourceFileLoader object at 0x7fb1ce8da570>
        __name__   = 'responses'
        __package__ = 'responses'
        __path__   = ['/home/runner/work/jsonschema-spec/jsonschema-spec/.venv/lib/python3.12/site-packages/responses']
        __spec__   = ModuleSpec(name='responses', loader=<_frozen_importlib_external.SourceFileLoader object at 0x7fb1ce8da570>, origin='/h...le_search_locations=['/home/runner/work/jsonschema-spec/jsonschema-spec/.venv/lib/python3.12/site-packages/responses'])
        _json_params_matcher = <function json_params_matcher at 0x7fb1ce76a980>
        _query_string_matcher = <function query_string_matcher at 0x7fb1ce76ab60>
        _urlencoded_params_matcher = <function urlencoded_params_matcher at 0x7fb1ce76a8e0>
        client     = <module 'http.client' from '/opt/hostedtoolcache/Python/3.12.0-beta.3/x64/lib/python3.12/http/client.py'>
        groupby    = <class 'itertools.groupby'>
        inspect    = <module 'inspect' from '/opt/hostedtoolcache/Python/3.12.0-beta.3/x64/lib/python3.12/inspect.py'>
        json_module = <module 'json' from '/opt/hostedtoolcache/Python/3.12.0-beta.3/x64/lib/python3.12/json/__init__.py'>
        logging    = <module 'logging' from '/opt/hostedtoolcache/Python/3.12.0-beta.3/x64/lib/python3.12/logging/__init__.py'>
        matchers   = <module 'responses.matchers' from '/home/runner/work/jsonschema-spec/jsonschema-spec/.venv/lib/python3.12/site-packages/responses/matchers.py'>
        namedtuple = <function namedtuple at 0x7fb1d09b4ea0>
        overload   = <function overload at 0x7fb1d084f240>
        partialmethod = <class 'functools.partialmethod'>
        registries = <module 'responses.registries' from '/home/runner/work/jsonschema-spec/jsonschema-spec/.venv/lib/python3.12/site-packages/responses/registries.py'>
        warn       = <built-in function warn>
        wraps      = <function wraps at 0x7fb1d09b6200>
        yaml       = <module 'yaml' from '/home/runner/work/jsonschema-spec/jsonschema-spec/.venv/lib/python3.12/site-packages/yaml/__init__.py'>
.venv/lib/python3.12/site-packages/typing_extensions.py:1174: in <module>
    class TypeVar(typing.TypeVar, _DefaultMixin, _root=True):
E   TypeError: type 'typing.TypeVar' is not an acceptable base type
```

Chain reaction with `responses` and `typing-extensions`.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -57,6 +57,7 @@ flynt = "1.0.0"
 mypy = "^1.4.1"
 types-PyYAML = "^6.0.12"
 types-requests = "^2.31.0"
+typing-extensions = "^4.6.0"  # required by responses. See https://github.com/p1c2u/jsonschema-spec/issues/44
 responses = "^0.23.1"
 deptry = "^0.12.0"
 

